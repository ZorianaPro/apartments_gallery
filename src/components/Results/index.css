.loading {
  color: green
}

.error {
  color: red
}