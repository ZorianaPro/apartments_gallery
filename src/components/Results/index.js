import React from 'react';
import {useSelector, useDispatch} from "../../react-redux-hooks";
import Offer from '../../components/Offer';
import allActions from "../../actions";


const Results = () => {
	const {apartments, loading, error_msg} = useSelector(state => ({
		apartments: state.apartments.apartments || [],
		loading: state.apartments.loading,
		error_msg: state.apartments.error_msg
	})) || [];

	const dispatch = useDispatch();

	React.useEffect(() => {
		dispatch(allActions.apartmentsActions.getApartments());
	}, []);

  if (loading) {
  	return (
  		<div className="loading">
			  Loading...
		  </div>
	  )
  }

	if (error_msg) {
		return (
			<div>
				{error_msg}
			</div>
		)
	}
	return (
		apartments.map((offer) => {
			return (
				<Offer id={offer.id}
				       name={offer.details.name}
				       detiles={offer.details}
				       photos={offer.photos}
				       price={offer.price}
				       rating={offer.rating}

				/>
			)
		})
	)
};

export default Results;