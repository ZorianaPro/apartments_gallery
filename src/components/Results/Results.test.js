import React, {useEffect} from 'react';
import ReactDOM from 'react-dom';
import Results from '../Results';
import { Provider } from 'react-redux';
import configureStore from "redux-mock-store";
import { shallow, configure, mount } from 'enzyme';
import jest from 'jest-mock';
import * as ReactReduxHooks from '../../react-redux-hooks';
import allActions from '../../actions';
import renderer from 'react-test-renderer'

describe('Results', () => {
	const INITIAL_STATE = {
		loading: false,
		apartments: [],
		error_msg: ''
	};
	let wrapper, store, useEffect;

	const mockStore = configureStore();

	const mockUseEffect = () => {
		useEffect.mockImplementationOnce(() => {
			ReactReduxHooks.useDispatch(allActions.apartmentsActions.getApartments)
		});
	};

	beforeEach(() => {
		store = mockStore(INITIAL_STATE);
		wrapper = mount(<Provider store={store}><Results /></Provider>);
		useEffect = jest.spyOn(React, "useEffect");
		mockUseEffect(); // important to do it twice
		mockUseEffect();
	});

	test('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
});

