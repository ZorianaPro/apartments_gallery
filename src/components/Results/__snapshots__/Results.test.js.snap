exports[`Results renders as expected 1`] = `null`;
